### PR TITLE
Update to switch to GA v2.0 endpoint

### LINKED ISSUE #12824: [formrecognizer] switch to testing GA endpoint

### PR DESCRIPTION
**Pull request by samvaity**
- Update recordings for 2.0 endpoint
- Autorest updates code-gen
- Update release version for FR lib to 3.0.0

Closes #12824

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Release History
 
-## 1.0.0-beta.5 (Unreleased)
+## 3.0.0 (Unreleased)
+First stable release of the azure-ai-formrecognizer package. Targets the service's v2.0 API.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: do you plan to update this in a second PR once we get confirmation?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:4]: I think I will keep this around until we get a confirmation. Don't want to merge this until we know for sure!

**Pull request merged**